fragmenting_writer.go

```diff
@@ -42,19 +42,20 @@ const (
 // for contents, a running checksum, and placeholders for the fragment flags
 // and final checksum value
 type writableFragment struct {
-	flagsRef    typed.ByteRef
-	checksumRef typed.BytesRef
-	checksum    Checksum
-	contents    *typed.WriteBuffer
-	frame       *Frame
+	flagsRef            typed.ByteRef
+	checksumRef         typed.BytesRef
+	checksum            Checksum
+	contents            *typed.WriteBuffer
+	frame               *Frame
+	dontReleaseChecksum bool
```

Comment:
This is a bit of an odd behaviour, some questions:
 * is there another way to do this without the if. e.g., can we move the responsibility of releasing the checksum to the caller of the non-relay codepath?
 * alternative to adding a bool + if check, what if we provide a decorator for `Checksum` that passes through all calls but `Reset`()`. Then when we create the `writableFragment`, we just wrap the checksum in our `noreleaseChecksum` type.